sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/IncrementalExecution.scala

```diff
@@ -50,6 +52,19 @@ class IncrementalExecution(
       stateStrategy)
 
   /**
+   * See [SPARK-18339]
+   * Walk the optimized logical plan and replace CurrentBatchTimestamp
+   * with the desired literal
+   */
+  override lazy val optimizedPlan: LogicalPlan = {
+    sparkSession.sessionState.optimizer.execute(withCachedData) transformAllExpressions {
+      case CurrentBatchTimestamp(timestamp) =>
```

Comment:
then following my comment above on the `CurrentBatchTimestamp` class, I would rewrite this as:
```scala
case ts @ CurrentBatchTimestamp(timestamp) =>
  logInfo(s"Current batch timestamp = $timestamp")
  ts.toLiteral
```